### PR TITLE
chore(rpc): support workers, file chooser, browser server

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/rpc/channels.ts [hunk 1]
```diff
@@ -46,7 +62,10 @@ export type BrowserInitializer = {};
 
 export interface BrowserContextChannel extends Channel {
   on(event: 'bindingCall', callback: (params: BindingCallChannel) => void): this;
+  on(event: 'close', callback: () => void): this;
   on(event: 'page', callback: (params: PageChannel) => void): this;
+  on(event: 'route', callback: (params: { route: RouteChannel, request: RequestChannel }) => void): this;
```
Comment (dgozman) [inline src/rpc/channels.ts:67]: Why do we have route event on both Page and BrowserContext?

File: src/rpc/client/browserServer.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    return request._object;
+  }
+
+  constructor(connection: Connection, channel: BrowserServerChannel, initializer: BrowserServerInitializer) {
+    super(connection, channel, initializer);
+    channel.on('close', () => this.emit(Events.BrowserServer.Close));
+  }
+
+  process(): ChildProcess {
+    return { pid: this._initializer.pid } as any;
```
Comment (dgozman) [inline src/rpc/client/browserServer.ts:34]: Oh my. Let's expose pid directly?

File: test/test.js [hunk 3]
```diff
@@ -24,6 +24,8 @@ const { Connection } = require('../lib/rpc/connection');
 const { helper } = require('../lib/helper');
 const { BrowserTypeDispatcher } = require('../lib/rpc/server/browserTypeDispatcher');
 
+Error.stackTraceLimit = 15;
```
Comment (dgozman) [inline test/test.js:27]: Why this?

**Review by dgozman — APPROVED** [review]

File: src/rpc/client/browserServer.ts (hunk 2 above)
Comment (pavelfeldman) [reply src/rpc/client/browserServer.ts:34]: That's for the test!

File: test/test.js (hunk 3 above)
Comment (pavelfeldman) [reply test/test.js:27]: That's because you like to assert stack against __filename.

File: src/rpc/channels.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/rpc/channels.ts:67]: Let me see.

**Pull request merged**